cmd/zrok/accessPrivate.go

```diff
@@ -85,40 +80,76 @@ func (cmd *accessPrivateCommand) run(_ *cobra.Command, args []string) {
 	}
 	logrus.Infof("allocated frontend '%v'", accessResp.Payload.FrontendToken)
 
-	cfg := privateFrontend.DefaultConfig("backend")
-	cfg.ShrToken = shrToken
-	cfg.Address = cmd.bindAddress
-	cfg.RequestsChan = make(chan *endpoints.Request, 1024)
-
-	c := make(chan os.Signal)
-	signal.Notify(c, os.Interrupt, syscall.SIGTERM)
-	go func() {
-		<-c
-		cmd.destroy(accessResp.Payload.FrontendToken, zrd.Env.ZId, shrToken, zrok, auth)
-		os.Exit(0)
-	}()
+	protocol := "http://"
+	switch accessResp.Payload.BackendMode {
+	case "tcpTunnel":
+		protocol = "tcp://"
+	}
 
-	frontend, err := privateFrontend.NewHTTP(cfg)
+	endpointUrl, err := url.Parse(protocol + cmd.bindAddress)
 	if err != nil {
 		if !panicInstead {
-			tui.Error("unable to create private frontend", err)
+			tui.Error("invalid endpoint address", err)
 		}
 		panic(err)
 	}
 
-	go func() {
-		if err := frontend.Run(); err != nil {
+	requests := make(chan *endpoints.Request, 1024)
+	if accessResp.Payload.BackendMode == "tcpTunnel" {
+		fe, err := tcpTunnel.NewFrontend(&tcpTunnel.FrontendConfig{
+			BindAddress:  cmd.bindAddress,
+			IdentityName: "backend",
+			ShrToken:     args[0],
+			RequestsChan: requests,
+		})
+		if err != nil {
+			if !panicInstead {
+				tui.Error("unable to create private frontend", err)
+			}
+			panic(err)
+		}
+		go func() {
```

Comment:
Probably not in this pr, but if more backend modes to come perhaps interface the frontends and move the running outside the configuration block?

Reply:
Sure. There isn't currently a standardized interface for frontend implementations... would need to add one (even if that's just a `Runner`/`Run` method).